In [1]:
# pip install pyspark

In [2]:
# PySpark core
from pyspark.sql import SparkSession

# Spark functions
import pyspark.sql.functions as F

Apache Spark digunakan karena kemampuannya dalam memproses data dalam skala besar secara terdistribusi. Spark memungkinkan pemrosesan data dilakukan secara paralel sehingga dapat meningkatkan efisiensi komputasi dibandingkan pendekatan tradisional yang hanya menggunakan satu prosesor. Dengan memanfaatkan SparkSession sebagai titik awal pemrosesan data, seluruh operasi manipulasi dan analisis data dapat dijalankan secara terintegrasi dalam satu lingkungan komputasi.

In [3]:
spark = SparkSession.builder \
    .appName("TMDB Data Loading") \
    .getOrCreate()

Kode ini berfungsi untuk mengaktifkan SparkSession, yang merupakan pintu masuk utama untuk menjalankan perintah Apache Spark. Nama aplikasi "TMDB Data Loading" diberikan agar proses pemrosesan data mudah diidentifikasi dalam sistem monitoring. Dengan menggunakan .getOrCreate(), sistem secara cerdas akan membuat sesi baru atau menggunakan sesi yang sudah berjalan jika ada. Dalam konteks Big Data, langkah ini sangat penting karena inilah yang memungkinkan data film yang besar diproses secara paralel di dalam memori, sehingga jauh lebih cepat dibandingkan pengolahan data tradisional.

In [4]:
file_path = "../data/raw/tmdb_movies_raw.parquet"

movies_df = spark.read.parquet(file_path)

Karena dataset sudah dalam format Parquet, Spark dapat membaca data dengan sangat cepat.

In [5]:
movies_df.printSchema()

root
 |-- Movie Name: string (nullable = true)
 |-- Year: string (nullable = true)
 |-- Certificate: string (nullable = true)
 |-- Duration: string (nullable = true)
 |-- Genre: string (nullable = true)
 |-- TMDB Rating: double (nullable = true)
 |-- Votes: long (nullable = true)
 |-- Director: string (nullable = true)
 |-- Stars: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- Grossed in $: string (nullable = true)
 |-- Plot: string (nullable = true)
 |-- Original Language: string (nullable = true)
 |-- Popularity: double (nullable = true)
 |-- TMDB ID: long (nullable = true)
 |-- IMDb ID: string (nullable = true)



Pemeriksaan struktur skema dilakukan untuk memahami tipe data dari setiap atribut yang terdapat dalam dataset.

In [6]:
movies_df.show(5)

+--------------------+----+-----------+--------+-------------------+-----------+-----+--------------------+--------------------+------------+--------------------+-----------------+----------+-------+---------+
|          Movie Name|Year|Certificate|Duration|              Genre|TMDB Rating|Votes|            Director|               Stars|Grossed in $|                Plot|Original Language|Popularity|TMDB ID|  IMDb ID|
+--------------------+----+-----------+--------+-------------------+-----------+-----+--------------------+--------------------+------------+--------------------+-----------------+----------+-------+---------+
|The Shawshank Red...|1994|          R| 142 min|       Drama, Crime|      8.716|29897|      Frank Darabont|[Tim Robbins, Mor...| $28,341,469|Imprisoned in the...|               en|   45.6702|    278|tt0111161|
|       The Godfather|1972|          R| 175 min|       Drama, Crime|      8.688|22579|Francis Ford Coppola|[Marlon Brando, A...|$245,066,411|Spanning the year..

Langkah ini bertujuan untuk memverifikasi bahwa data telah berhasil dimuat ke dalam lingkungan Apache Spark serta memberikan gambaran awal mengenai struktur dan isi dataset.

In [7]:
movies_df.count()

9752

Dataset yang digunakan berisi 9752 film yang dikumpulkan dari platform TMDB. Jumlah film tidak 10.000 karena sudah dilakukan deduplikasi sebelum mengunduh raw dataset.